### PR TITLE
Lfmcmc rand engine

### PR DESCRIPTION
**Pull request by gvegayon**
The LFMCMC function no longer needs an `epiworld_model` object to be instantiated. A rand engine is automatically created when `NULL` is passed instead. The data type was changed to double (from int) to give more flexibility. I added a new test that uses LFMCMC to estimate the mean and standard deviation of a normal vector. Finally, I modified the vignette to have a better kernel function (does a better job now!).

### Function Enhancements:
* Modified the `LFMCMC` function to accept a `NULL` model, creating a new random-number generator engine if no model is provided. (`R/LFMCMC.R`, `man/LFMCMC.Rd`) [[1]](diffhunk://#diff-7831e522d023edbea0571027e997624dbac7397dd1b9b7c8ad5675b1b42753e0L76-R90) [[2]](diffhunk://#diff-c64236ccce5174dbd0e931a3e61a5e7d7ba6f5cc266e9839aeba2fbdc3792e14L21-R21)
* Updated the `run_lfmcmc` function to include parameter type conversions and ensure compatibility with the new `LFMCMC` implementation. (`R/LFMCMC.R`)

### C++ Code Updates:
* Changed the data type in the C++ code from `int` to `double` for better precision and compatibility with the new LFMCMC features. (`src/cpp11.cpp`, `src/lfmcmc.cpp`) [[1]](diffhunk://#diff-76799f7c917e848b3575d571772e1e9cd7d6f06f3f58fcb2536295dcccc6f768L401-R411) [[2]](diffhunk://#diff-ddf82580217fd5f089f2ce1a632b0cb3775464f9b18667c2cc412f0e5e6b57b0L45-R52)
* Added logic to the `LFMCMC_cpp` function to handle the creation of a new random-number generator engine when the model is `NULL`. (`src/lfmcmc.cpp`)

### Testing and Documentation:
* Adds new test to file `test-lfmcmc.R` to validate the functionality of the LFMCMC algorithm with different models and ensure accuracy. (`inst/tinytest/test-lfmcmc.R`)
* Updated the vignette `likelihood-free-mcmc.Rmd` to reflect changes in the LFMCMC setup and usage examples. (`vignettes/likelihood-free-mcmc.Rmd`) [[1]](diffhunk://#diff-2c82bc39ffa89e0a5bc6853809ede6fb01a702ebed40f78a176b27648d71c8c4L35-R35) [[2]](diffhunk://#diff-2c82bc39ffa89e0a5bc6853809ede6fb01a702ebed40f78a176b27648d71c8c4L61-R61)
* Adds Andrew Pulsipher as co-author of package

These changes collectively enhance the `epiworldR` package by providing more robust and flexible tools for model parameter estimation and simulation.

### CODE REVIEW

**Review by apulsipher — CHANGES_REQUESTED** [review]
I love the purpose of this PR to allow instantiation without a model. I've two questions/changes to address here. 

Additionally, the PR is failing most of the checks. Changing the data type from `int` to `double` causes unit tests to fail and causes the examples at the top of the `.R` file to throw errors. Please fix those before merging!

File: vignettes/likelihood-free-mcmc.Rmd [hunk 1]
```diff
@@ -95,7 +95,7 @@ kernelfun <- function(stats_now, stats_obs, epsilon) {
     stats_obs,
     stats_now))
```
Comment (gvegayon) [inline vignettes/likelihood-free-mcmc.Rmd:96]: Remove this

**Review by apulsipher — APPROVED** [review]
I resolved the issues we discussed. @gvegayon Can you check that I've added myself as author correctly before merging?

**Comment by gvegayon** [comment]
> I resolved the issues we discussed. @gvegayon Can you check that I've added myself as author correctly before merging?

LGTM!

**Pull request merged**